cms/djangoapps/contentstore/utils.py

```diff
@@ -582,55 +582,91 @@ def is_self_paced(course):
     return course and course.self_paced
 
 
-def get_sibling_urls(subsection):
+def get_sibling_urls(subsection, unit_location):
     """
     Given a subsection, returns the urls for the next and previous units.
 
     (the first unit of the next subsection or section, and
     the last unit of the previous subsection/section)
     """
+    # pylint: disable=too-many-statements
     section = subsection.get_parent()
     prev_url = next_url = ''
     prev_loc = next_loc = None
-    last_block = None
-    siblings = list(section.get_children())
-    for i, block in enumerate(siblings):
-        if block.location == subsection.location:
-            if last_block:
-                try:
-                    prev_loc = last_block.get_children()[0].location
-                except IndexError:
-                    pass
+
+    unit_index = subsection.children.index(unit_location)
+    try:
```

Comment:
Most of code blocks look similar something like this can be done to remove duplication
```suggestion
def get_sibling_index(index:int):
        try:
            return subsection.children[unit_index]
        except IndexError:
            return None
prev_loc = get_sibling_index(unit_index-1)
next_loc = get_sibling_index(unit_index+1)
```